### PR TITLE
Update SDL2, json, tremor, freetype, harfbuzz, libmpg123, libsamplerate, fluidsynth and fluidlite

### PR DESCRIPTION
**Pull request by fdelapena**
Untested. Assuming some patches, if any, could require some amendments.

### CODE REVIEW

File: shared/packages.sh [hunk 1]
```diff
@@ -11,13 +11,13 @@ LIBPNG_URL="https://download.sourceforge.net/libpng/$lib-$ver.tar.xz"
...
 lib=freetype
-ver=2.10.4
+ver=2.11.0
 FREETYPE_URL="https://download.savannah.gnu.org/releases/$lib/$lib-$ver.tar.xz"
 FREETYPE_DIR="$lib-$ver"
 FREETYPE_ARGS="--without-bzip2 --without-png --without-zlib"
 
 lib=harfbuzz
-ver=2.8.2
+ver=2.9.0
```
Comment (carstene1ns) [inline shared/packages.sh:20]: We might want to migrate to meson or cmake for harfbuzz.

**Review by carstene1ns — COMMENTED** [review]
Will do a test run for 3ds/wii/switch and vita.
The rest should be trivial. Harfbuzz patches might not apply cleanly though.

File: shared/packages.sh (hunk 1 above)
Comment (Ghabry) [reply shared/packages.sh:20]: the same kinda applies to pixman. They are also pushing meson, so we cannot ignore this for long anymore ^^'

**Comment by Ghabry** [comment]
did you mess up a rebase? We already have these commits.

**Comment by fdelapena** [comment]
> did you mess up a rebase? We already have these commits.

Fixed, I forgot to rebase from local master properly

**Comment by Ghabry** [comment]
Could you make a buildtest of Player using these buildscripts, at least on Linux? Then you can see at least if the patches still apply

**Comment by carstene1ns** [comment]
Sorry, had no time to do the testing for the toolchains yet.

**Comment by Ghabry** [comment]
Can you bump:

json 3.10.4
samplerate 0.2.2
FluidLite https://github.com/divideconcept/FluidLite/commit/57a0e74e708f699b13d7c85b28a4e1ff5b71887c

**Comment by Ghabry** [comment]
Made a build test of Player and everything builds here. Just apply my patch and delete the unused files and we are good to go :+1:

**Comment by Ghabry** [comment]
for some reason CMake installs now in lib64 for linux-static, including the pkg-config files. o_O

Added a temporary workaround to ensure this still compiles but should be fixed properly

**Pull request merged**